core/cmd/on_chain_config_cmd.go

```
@@ -0,0 +1,399 @@
+		}
+		valueType = knownType
+	} else if !force && knownType != valueType {
+		return setArgs{}, RiverError(Err_INVALID_ARGUMENT, "type mismatch for key", "key", key, "known_type", knownType, "provided_type", valueType)
+	}
+
+	blockNum, err := strconv.ParseUint(blockNumStr, 10, 64)
+	if err != nil {
+		return setArgs{}, err
+	}
+
+	valueBytes, err := encodeValue(valueType, value)
+	if err != nil {
+		return setArgs{}, err
+	}
+
+	return setArgs{
+		key:      key,
+		blockNum: blockNum,
+		value:    valueBytes,
+	}, nil
+}
+
+func submitConfig(ctx context.Context, cfg *config.Config, args []setArgs) error {
+	wallet, err := crypto.NewWalletFromEnv(ctx, "PRIVATE_KEY")
+	if err != nil {
+		return err
+	}
+
+	blockchain, err := crypto.NewBlockchain(
+		ctx,
+		&cfg.RiverChain,
+		wallet,
+		infra.NewMetricsFactory(nil, "river", "cmdline"),
+		nil,
+	)
+	if err != nil {
+		return err
+	}
+	blockchain.StartChainMonitor(ctx)
+
+	caller, err := river.NewRiverConfigV1Transactor(cfg.RegistryContract.Address, blockchain.Client)
+	if err != nil {
+		return err
+	}
+
+	ctx, cancel := context.WithTimeout(ctx, 20*time.Second)
+	defer cancel()
+
+	var txs []crypto.TransactionPoolPendingTransaction
+	for _, arg := range args {
+		fmt.Printf("Setting %s to %s on block %d\n", arg.key, hex.EncodeToString(arg.value), arg.blockNum)
+		tx, err := blockchain.TxPool.Submit(
+			ctx,
+			"SetConfiguration",
+			func(opts *bind.TransactOpts) (*types.Transaction, error) {
+				return caller.SetConfiguration(
+					opts,
+					crypto.HashSettingName(arg.key),
+					arg.blockNum,
+					arg.value,
+				)
+			},
+		)
+		if err != nil {
+			return err
+		}
+		txs = append(txs, tx)
+	}
+
+	for _, tx := range txs {
+		receipt, err := tx.Wait(ctx)
+		if err != nil {
+			return err
+		}
+		if receipt.Status != types.ReceiptStatusSuccessful {
+			return RiverError(Err_INTERNAL, "transaction failed", "tx", receipt.TxHash)
```

Maybe capture all transaction results and return a summary instead of returning immediately so there is an overview which transaction did fail and which pass.